internal/cache/theme.go

```diff
@@ -0,0 +1,64 @@
+package cache
+
+import "fyne.io/fyne/v2"
+
+var (
+	forWidgets = make(map[fyne.Resource]fyne.Widget)
```

Comment:
Is there a way to do without this forWidget map, it seems a bit weird to me and like it's an inversion of encapsulation in a way. Since ThemedResource is a widget, is it not just taken care of by the overrides map?

Anyway, there's another bigger issue with ThemedResources with this (see other comment) and in reworking for that we may find a solution for this as well

Reply:
The point above is interesting, though based on "Since ThemedResource is a widget" which is false unfortunately. I think the point below did open up a way to remove this breaking of encapsulation.